extensions/amp-analytics/0.1/requests.js

```diff
@@ -325,6 +369,25 @@ export class RequestHandler {
       this.refreshBatchInterval_();
     }, interval);
   }
+
+  /**
+   * Composes a request URL given a base and requestOrigin
+   * @private
+   * @param {string} baseUrl
+   * @param {string=} opt_requestOrigin
+   * @return {string}
+   */
+  composeRequestUrl_(baseUrl, opt_requestOrigin) {
+    if (opt_requestOrigin) {
+      // parse can handle relative request origins and will use the
+      // current page's origin if so. However, we only want to accept absolute
+      // URLs for request origin to keep our URL composition rules simple
+      const requestOriginInfo = this.urlService_.parse(opt_requestOrigin);
+      return requestOriginInfo.origin + baseUrl;
```

Comment:
Can we please make a comment, that amp-analytics simply respect the requestOrigin and baseUrl. There's no check and the request will fail silently if they don't compose a valid url.